modules/library/main/src/main/java/org/geotools/styling/AbstractSymbolizer.java

```diff
@@ -149,9 +149,9 @@ public boolean equals(Object obj) {
         if (options == null) {
             if (other.options != null && !other.options.isEmpty())
                 return false;
-        } else if (!options.equals(other.options))
-            if (options.isEmpty() && other.options != null) 
-                return false;
+        } else if (!options.equals(other.options)) {
+            return false;
+        }
         return true;
```

Comment:
This is a very confusing piece of code, so before we had this:
```
if (options == null) {
	if (other.options != null && !other.options.isEmpty())
		return false;
} else if (!options.equals(other.options))
	if (options.isEmpty() && other.options != null) 
		return false;
}
```
and now we have this:
```
if (options == null) {
	if (other.options != null && !other.options.isEmpty())
		return false;
} else if (!options.equals(other.options)) {
	return false;
}
```

I think that old behavior and the new one differ, let's consider this situation:
```
This options and other options are not the same object,
but this options is empty and the other options is NULL.
```
It seems to me that the old code would not return ``FALSE`` but the new code will. I'm not sure if this was the indented behavior or not.

If the intention of this code is to consider two maps equal if the ``equals(...)`` method returns ``TRUE`` or if both are ``NULL`` or empty then I would have wrote something like this:
```
if (options == null || options.isEmpty()) {
	// this options are NULL or empty
	if (other.options != null && !other.options.isEmpty()) {
		// the other options are neither NULL or empty
		return false;
	}
} else if (!options.equals(other.options)) {
	// options are not considered the same
	return false;
}
```